sdk/python/feast/infra/offline_stores/contrib/trino_offline_store/trino_source.py

```diff
@@ -0,0 +1,241 @@
+from typing import Callable, Dict, Iterable, Optional, Tuple
+
+from feast import ValueType
+from feast.data_source import DataSource
+from feast.infra.offline_stores.contrib.trino_offline_store.trino_queries import Trino
+from feast.infra.offline_stores.contrib.trino_offline_store.trino_type_map import (
+    trino_to_feast_value_type,
+)
+from feast.protos.feast.core.DataSource_pb2 import DataSource as DataSourceProto
+from feast.protos.feast.core.SavedDataset_pb2 import (
+    SavedDatasetStorage as SavedDatasetStorageProto,
+)
+from feast.repo_config import RepoConfig
+from feast.saved_dataset import SavedDatasetStorage
+
+
+class TrinoOptions:
+    """
+    DataSource Trino options used to source features from Trino query
+    """
+
+    def __init__(self, table: Optional[str], query: Optional[str]):
+        self._table = table
+        self._query = query
+
+    @property
+    def query(self):
+        """
+        Returns the Trino SQL query referenced by this source
+        """
+        return self._query
+
+    @query.setter
+    def query(self, query):
+        """
+        Sets the Trino SQL query referenced by this source
+        """
+        self._query = query
+
+    @property
+    def table(self):
+        """
+        Returns the table ref of this Trino table
+        """
+        return self._table
+
+    @table.setter
+    def table(self, table):
+        """
+        Sets the table ref of this Trino table
+        """
+        self._table = table
+
+    @classmethod
+    def from_proto(cls, trino_options_proto: DataSourceProto.TrinoOptions):
+        """
+        Creates a TrinoOptions from a protobuf representation of a Trino option
+        Args:
+            trino_options_proto: A protobuf representation of a DataSource
+        Returns:
+            Returns a TrinoOptions object based on the trino_options protobuf
+        """
+        trino_options = cls(
+            table=trino_options_proto.table, query=trino_options_proto.query,
+        )
+
+        return trino_options
+
+    def to_proto(self) -> DataSourceProto.TrinoOptions:
+        """
+        Converts an TrinoOptionsProto object to its protobuf representation.
+        Returns:
+            TrinoOptionsProto protobuf
+        """
+
+        trino_options_proto = DataSourceProto.TrinoOptions(
+            table=self.table, query=self.query,
+        )
+
+        return trino_options_proto
+
+
+class TrinoSource(DataSource):
+    def __init__(
+        self,
+        event_timestamp_column: Optional[str] = "",
```

Comment:
```suggestion
    def __init__(
        self,
        *,
        event_timestamp_column: Optional[str] = "",
```

This will force all args to be specified as kwargs, which is what we want to move to entirely.

Reply:
Good idea. Done